src/rgw/motr/synchronization/rgw_motr_gc_sync.cc

```diff
@@ -0,0 +1,77 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * SAL implementation for the CORTX Motr backend
+ *
+ * Copyright (C) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "motr/synchronization/rgw_motr_gc_sync.h"
+
+int MotrGCThreadSync::lock(const std::string &lock_name, MotrLockType lock_type, utime_t lock_duration, const std::string locker_id) {
+  assert(motr_gclock);
```

Comment:
why are we using `assert`? 
simple `if-else` should work, right?

Reply:
assert is for developer's test purpose and should ideally be enabled only for debug builds. However, this, would be removed, as Ceph doesn't have mechanism in it's code base today that could be enabled only in debug builds. So, will remove it. Will use "if" for parameter validations